update es to support muti platform

linuzb opened the pull request:
1. update elasticsearch to 7.x
2. fix container status
3. add docker-build-push pipeline for mutiplatform

hnhbwlp commented:
Why does it need to modify the index type? What is the difference between _doc and data?

linuzb commented:
> Why does it need to modify the index type? What is the difference between _doc and data?

@hnhbwlp They are all mapping types. [What are mapping types?](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html#_what_are_mapping_types)

Custom mapping types were supported in Elasticsearch before version 5.x . You could have multiple mapping types in a single index. 

However, starting from version 6.x, an index only supports one mapping type, which can still be customized. [Why are mapping types being removed?](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html#_why_are_mapping_types_being_removed)

In version 7.x, the default type is _doc (custom types can still be used, but it is not recommended). Creating a mapping with a specified mapping type is not supported in version 7.x. Additionally, the new index API format in version 7.x is PUT /{index}/_doc/{id} and POST {index}/_doc. It is important to note that for the sake of API compatibility, _doc is not a type but a permanent part of the API request path.


Therefore, in the provided GitHub link, I removed the _doc type in mapping creation and replaced all custom data with _doc. 

Starting from Elasticsearch 8.x, custom types are no longer supported.

larryck commented:
It's LGTM. 

> > Why does it need to modify the index type? What is the difference between _doc and data?
> 
> @hnhbwlp They are all mapping types. [What are mapping types?](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html#_what_are_mapping_types)
> 
> Custom mapping types were supported in Elasticsearch before version 5.x . You could have multiple mapping types in a single index.
> 
> However, starting from version 6.x, an index only supports one mapping type, which can still be customized. [Why are mapping types being removed?](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html#_why_are_mapping_types_being_removed)
> 
> In version 7.x, the default type is _doc (custom types can still be used, but it is not recommended). Creating a mapping with a specified mapping type is not supported in version 7.x. Additionally, the new index API format in version 7.x is PUT /{index}/_doc/{id} and POST {index}/_doc. It is important to note that for the sake of API compatibility, _doc is not a type but a permanent part of the API request path.
> 
> Therefore, I removed the _doc type in mapping creation and replaced all custom data with _doc.
> 
> Starting from Elasticsearch 8.x, custom types are no longer supported.

It's LGTM. We need to upgrade to newer version of es in the long run. As far as I know, this PR has run properly on several arm machines.
